pennylane/interfaces/jax_jit_tuple.py

```diff
@@ -0,0 +1,297 @@
+            shape_dtypes.append(shape_and_dtype)
+        else:
+            if num_measurements == 1:
+                s = [shape_and_dtype for _ in range(len(t.trainable_params))]
+                shape_dtypes.append(tuple(s))
+            else:
+                s = [tuple(_s for _ in range(len(t.trainable_params))) for _s in shape_and_dtype]
+                shape_dtypes.append(tuple(s))
+
+    if len(tapes) == 1:
+        return shape_dtypes[0]
+
+    return tuple(shape_dtypes)
+
+
+def _squeeze_nested(jvps, multi_measurements, multi_params):
+    """Post-process the JVP by squeezing arrays in the result with a nested structure."""
+    res_jvps = []
+    for m, p, j in zip(multi_measurements, multi_params, jvps):
+        if m and p:
+            res_jvps.append(tuple(jnp.squeeze(j_comp) for j_comp in j))
+        else:
+            res_jvps.append(jnp.squeeze(j))
+    return res_jvps
+
+
+def execute_tuple(tapes, device, execute_fn, gradient_fn, gradient_kwargs, _n=1, max_diff=1):
+    """Execute a batch of tapes with JAX parameters on a device.
+
+    Args:
+        tapes (Sequence[.QuantumTape]): batch of tapes to execute
+        device (.Device): Device to use to execute the batch of tapes.
+            If the device does not provide a ``batch_execute`` method,
+            by default the tapes will be executed in serial.
+        execute_fn (callable): The execution function used to execute the tapes
+            during the forward pass. This function must return a tuple ``(results, jacobians)``.
+            If ``jacobians`` is an empty list, then ``gradient_fn`` is used to
+            compute the gradients during the backwards pass.
+        gradient_kwargs (dict): dictionary of keyword arguments to pass when
+            determining the gradients of tapes
+        gradient_fn (callable): the gradient function to use to compute quantum gradients
+        _n (int): a positive integer used to track nesting of derivatives, for example
+            if the nth-order derivative is requested.
+        max_diff (int): If ``gradient_fn`` is a gradient transform, this option specifies
+            the maximum order of derivatives to support. Increasing this value allows
+            for higher order derivatives to be extracted, at the cost of additional
+            (classical) computational overhead during the backwards pass.
+
+    Returns:
+        list[list[float]]: A nested list of tape results. Each element in
+        the returned list corresponds in order to the provided tapes.
+    """
+    # pylint: disable=unused-argument
+    if max_diff > 1:
+        raise InterfaceUnsupportedError(
+            "The JAX-JIT interface only supports first order derivatives."
+        )
+
+    if any(
+        m.return_type in (qml.measurements.Counts, qml.measurements.AllCounts)
+        for t in tapes
+        for m in t.measurements
+    ):
+        raise NotImplementedError("The JAX-JIT interface doesn't support qml.counts.")
```

Comment:
Can we not just skip it like we do in other interface? Or is it a Jit limitation.

Reply:
Obtaining the shape for `qml.counts` is not implemented and raises:
> E       pennylane.QuantumFunctionError: Cannot deduce the shape of the measurement process with unrecognized return_type ObservableReturnTypes.Counts.

I'll add a comment for this in the code.

Reply:
Thanks!